[DOCS-14342] Add convert timestamp for Databricks (#36592)
DataDog/documentation: content/en/observability_pipelines/destinations/databricks.md

After you select the Databricks (Zerobus) destination in the pipeline UI:

> <div class="alert alert-warning">Databricks Zerobus doesn't convert timestamps in string format to Databricks' <a href="https://docs.databricks.com/aws/en/sql/language-manual/data-types/timestamp-type"><code>TIMESTAMP</code> type</a>. If your table uses a timestamp column, see <a href="#convert-string-timestamps-to-timestamp-format">Convert string timestamps to timestamp format</a> for more information.</div>

Author: ```suggestion
<div class="alert alert-warning">Databricks (Zerobus) doesn't convert timestamps in string format to Databricks' <a href="https://docs.databricks.com/aws/en/sql/language-manual/data-types/timestamp-type"><code>TIMESTAMP</code> type</a>. If your table uses a timestamp column, see <a href="#convert-string-timestamps-to-timestamp-format">Convert string timestamps to timestamp format</a> for more information.</div>
```

{{% observability_pipelines/destination_buffer %}}

### Convert string timestamps to timestamp format

> If your logs have timestamps in string format and your Databricks table has a timestamp column declared as a [`TIMESTAMP` type][11], you must convert the string to timestamp format before sending logs to the Databricks (Zerobus) destination. Otherwise, the Worker throws an error similar to:

Reviewer: Similar note to above - "type" and "format" appear to be used interchangeably for `TIMESTAMP`, consider standardizing to one or the other

Author: Thanks @jeff-morgan-dd! I was mulling over this when I was writing it. So, Databricks is not able to convert a string timestamp into its `TIMESTAMP` type, so OP has to convert it to the timestamp format/object, which Databricks can convert to its `TIMESTAMP` type. Do you think I should explain that?

Reviewer: Ohh interesting! In that case yes, i think it's worth a quick line explaining that flow and that the format and type are distinct objects. Thanks for the clarification!

Author: @jeff-morgan-dd how does this look?

```suggestion
If your logs have timestamps in string format and your Databricks table has a timestamp column declared as a [`TIMESTAMP` type][11], you must convert the string to timestamp format before sending logs to the Databricks (Zerobus) destination. Databricks (Zerobus) can only convert the timestamp format to its `TIMESTAMP` type. 

If you do not convert the string timestamp, the Worker throws an error similar to:
```

Reviewer: I think it might be even clearer/more direct with a positive construction i.e. "Databricks (Zerobus) can only convert the timestamp format to its `TIMESTAMP` type."

Thoughts?

Author: Ah yes, that makes sense! Updated the suggestion.

```
Protobuf encoding failed: Error converting timestamp field: Can't convert '2012-04-23T10[41]15Z' to i64: invalid digit found in string
```

To convert timestamps in string format to timestamp format:

1. Add a [Custom Processor][12] to your pipeline.
1. Add a function with the following custom script:
    ```
    .timestamp = parse_timestamp!(.timestamp, format: "%+")
    ```

>     See [Parse timestamp][13] for more information.

Reviewer: The heading this links to is parse_timestamp - consider using that for the link text?

Author: ```suggestion
    See [parse_timestamp][13] for more information.
```
